src/libraries/Common/src/Interop/Browser/Interop.HttpHandlerService.cs

```diff
@@ -0,0 +1,535 @@
+
+internal static partial class Interop
+{
+    internal static partial class Browser
+    {
+        internal interface IHttpHandlerService
+        {
+            Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken);
+            void Dispose();
+        }
+
+        public class BrowserHttpHandlerService : IHttpHandlerService, IDisposable
+        {
+
+            private static JSObject? fetch;
+            private static JSObject? window;
+
+            /// <summary>
+            /// Gets whether the current Browser supports streaming responses
+            /// </summary>
+            private static bool StreamingSupported { get; }
+
+            public BrowserHttpHandlerService()
+            { }
+
+            static BrowserHttpHandlerService()
+            {
+                using (var streamingSupported = new Function("return typeof Response !== 'undefined' && 'body' in Response.prototype && typeof ReadableStream === 'function'"))
+                    StreamingSupported = (bool)streamingSupported.Call();
+                handlerInit();
+            }
+
+            private static void handlerInit()
+            {
+                window = (JSObject)Interop.Runtime.GetGlobalObject("window");
+                fetch = (JSObject)Interop.Runtime.GetGlobalObject("fetch");
+            }
+
+            public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                // There is a race condition on Safari as a result of using TaskCompletionSource that
+                // causes a stack exceeded error being thrown.  More information can be found here:
+                // https://devblogs.microsoft.com/premier-developer/the-danger-of-taskcompletionsourcet-class/
+                var tcs = new TaskCompletionSource<HttpResponseMessage>(TaskCreationOptions.RunContinuationsAsynchronously);
+                using (cancellationToken.Register(() => tcs.TrySetCanceled()))
+                {
+#pragma warning disable 4014
+                    doFetch(tcs, request, cancellationToken).ConfigureAwait(false);
+#pragma warning restore 4014
+
+                    return await tcs.Task.ConfigureAwait(true);
+                }
+            }
+
+            private async Task doFetch(TaskCompletionSource<HttpResponseMessage> tcs, HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                if (request == null)
+                {
+                    throw new ArgumentNullException(nameof(request));
+                }
+
+                try
+                {
+                    var requestObject = new JSObject();
+
+                    if (request.Properties.TryGetValue("WebAssemblyFetchOptions", out var fetchOoptionsValue) &&
+                        fetchOoptionsValue is IDictionary<string, object> fetchOptions)
+                    {
+                        foreach (var item in fetchOptions)
```

Comment:
var => whatever the type actually is... object?

Reply:
addressed in recent commits